src/persist-client/src/operators/shard_source.rs

```diff
@@ -341,7 +354,7 @@ where
             // with the handle, so even if we never make it to this block (e.g.
             // budget of usize::MAX), because the stream has no data, we don't
             // cause unbounded buffering in timely.
-            while inflight_bytes >= flow_control_max_inflight_bytes {
+            while inflight_bytes >= max_inflight_bytes {
```

Comment:
am I right that in the new structure, we can never get here if flow_control_bytes is None? if so, maybe toss in an `assert_eq!(flow_control_bytes, None)` with a (short) comment

Reply:
Yes, you are right. Added!